core/src/libtx/build.rs

```diff
@@ -113,6 +115,72 @@ where
 	)
 }
 
+/// Adds multiple outputs with the provided value and key identifier from the
+/// keychain. Optimized by multiple threads.
+pub fn outputs<K>(amounts_and_keys: Vec<(u64, Identifier)>) -> Box<Append<K>>
```

Comment:
Maybe change the return type to `Result<Box<Append<K>>, Error>`, and use the `?` below to avoid panics in the caller.